plugin.rb

```diff
@@ -306,6 +311,180 @@ def account_created
           format.json { render json: success_json }
         end
       end
+
+      def create
```

Comment:
I think that it will be more easier simply create a new endpoint that will allow us to find address by zip code. 
Example:
- guest choose sign up with credit card
- guest provides first/last name details
- guest provides credit card details
- guest provides zip code details
- guest clicks on submit button
- we sending ajax request to find address by zip code
- guest sees error message if request fails
- we taking address from ajax request response, saving it in billing address custom user field and continue our current sign up process flow

In this case it will require a less changes in he code and we will not need to override user creation logic.